test/e2e/framework/pod/utils_test.go

```diff
@@ -65,7 +66,7 @@ func TestMixinRestrictedPodSecurity(t *testing.T) {
 	for _, pod := range restrictablePods {
 		t.Run(pod.Name, func(t *testing.T) {
 			p := pod // closure
-			assert.NoError(t, MixinRestrictedPodSecurity(&p))
+			require.NoError(t, MixinRestrictedPodSecurity(&p))
 			assert.Equal(t, GetRestrictedPodSecurityContext(), p.Spec.SecurityContext,
 				"Mixed in PodSecurityContext should equal the from-scratch PodSecurityContext")
 			assert.Equal(t, GetRestrictedContainerSecurityContext(), p.Spec.Containers[0].SecurityContext,
```

Comment:
what is the issue with using assert.NoError?
the remaining of the assertion is done with the assert package?

Reply:
that seems like some testify weirdness unless the whole assert package is a soft-assert. i would have assumed that assert.Error and NoError block further execution.

Reply:
it seems like so:

https://github.com/stretchr/testify/blob/master/assert/assertions.go#L1579

https://github.com/stretchr/testify/blob/master/assert/assertions.go#L348

it calls t.Errorf